include/seqan3/core/metafunction/iterator.hpp

```diff
@@ -65,7 +66,7 @@ template <std::InputIterator it_t>
 struct value_type<it_t>
 {
     //!\brief Return the member type as return type.
-    using type = ranges::value_type_t<it_t>;
+    using type = typename std::iterator_traits<std::remove_reference_t<it_t>>::value_type;
```

Comment:
As I understand std::iterator_traits is specialized with an Iterator type but you are passing a range type?
Shouldn't it be rather something like 
```suggestion
    using type = typename std::iterator_traits<std::remove_reference_t<decltype(it_t{}.begin())>>::value_type;
```